### PR TITLE
Adds support for parity integration tests with either fixed block time or instant seal.

### PR DESCRIPTION
**Pull request by MicahZoltu**
Speeds up iteration time when working on test infrastructure by enhancing what Docker images cash.

### CODE REVIEW

File: parity-dev-node/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ COPY start.sh /start.sh
 RUN echo "" > /parity/password
 RUN chmod +x /start.sh
 
-ENTRYPOINT /start.sh
+ENTRYPOINT [ "/start.sh" ]
```
Comment (adrake33) [inline parity-dev-node/Dockerfile]: Why did you add the brackets?

File: Dockerfile-integration-tests [hunk 2]
```diff
@@ -6,8 +6,13 @@ WORKDIR /app
 RUN npm install
 COPY tsconfig.json /app/tsconfig.json
 COPY typings/ /app/typings/
-COPY source/ /app/source/
+COPY source/contracts/ /app/source/contracts/
+COPY source/deployment/CompileContracts.ts /app/source/deployment/CompileContracts.ts
+COPY source/libraries/CompileSolidity.ts /app/source/libraries/CompileSolidity.ts
```
Comment (adrake33) [inline Dockerfile-integration-tests]: Why are you copying only these files and then copying everything in source later on?

Comment (MicahZoltu) [reply Dockerfile-integration-tests]: We only need contracts and these two files to compile the contracts and store the compiled results.  Docker will cache this and only recompile all of the contracts if one of the files copied above the contract compilation line changes.  This allows us to iterate on the integration tests without having to re-compile all of the contracts every run.

File: parity-dev-node/Dockerfile (hunk 1 above)
Comment (MicahZoltu) [reply parity-dev-node/Dockerfile]: Because Docker is stupid.  😢 `ENTRYPOINT /start.sh` is notably different from `ENTRYPOINT [ "/start.sh" ]`.  In this particular case, it needs to be the latter format to support passing in additional variables to the script and docker run time.

File: Dockerfile-integration-tests (hunk 2 above)
Comment (adrake33) [reply Dockerfile-integration-tests]: Ok, I figured it had to do with caching the contracts.  So Docker's cache works on a line-by-line basis?

Comment (MicahZoltu) [reply Dockerfile-integration-tests]: Yeah, each Docker line is another cache layer.  If you do something like this though, its considered one line:
```
RUN echo hello \
    && echo world
```

Comment (adrake33) [reply Dockerfile-integration-tests]: Yeah, mean I've seen it cache certain parts of the file that it's already run.  It just hadn't occurred to me that we could do the caching in this way so that if non-contract files are modified, it won't recompile.  Good idea!
